pithos/plugins/notification_icon.py

```diff
@@ -112,6 +79,18 @@ def check_for_plasma():
                 None
             )
 
+        def on_StatusNotifierWatcher_finish(source, result, data):
+            try:
+                StatusNotifierWatcher = Gio.DBusProxy.new_finish(result)
+            except GLib.Error as e:
+                logging.debug(e)
+                check_for_x()
+            else:
+                if StatusNotifierWatcher.get_name_owner():
+                    init_appindicator()
+                else:
+                    check_for_plasma()
```

Comment:
So to summarize if we wanted to re-implement AppIndicator3 in python we'd need to create 3 interfaces:

StatusNotifierHost
StatusNotifierItem
Dbusmenu

Basically aside from the Buntu's which already have AppIndicator3 available the only DE that it would actually work on is KDE Plasma right? Reading the docs for the spec it's a little suspect. For instance all the icon props have a redundant *Pixmap prop. It says that all props are mandatory yet it also says that it's not possible to set the normal icon prop and the *Pixmap prop at the same time.